staging/vhost-device-can/src/vhu_can.rs

```diff
@@ -483,42 +501,71 @@ impl VhostUserCanBackend {
         }
 
         for desc_chain in requests {
-            let descriptors: Vec<_> = desc_chain.clone().collect();
-
-            if descriptors.len() != 2 {
-                return Err(Error::UnexpectedDescriptorCount(descriptors.len()));
-            }
+            let atomic_mem = self.mem.as_ref().unwrap().memory();
 
-            let desc_request = descriptors[0];
-            if desc_request.is_write_only() {
+            let Ok(mut reader) = desc_chain.clone().reader(&atomic_mem) else {
+                trace!("Error::UnexpectedWriteOnlyDescriptor");
```

Comment:
Is `UnexpectedWriteOnlyDescriptor` the right error in this case?

Reply:
I think you can do something like:
```rust
    let mut reader = desc_chain
        .clone()
        .reader(&mem)
        .map_err(|_| Error::DescriptorReadFailed)?;
```

Reply:
Indeed! I simplified it as you described, thanks!